docs/book/src/developer/providers/migrations/v1.8-to-v1.9.md

```diff
@@ -0,0 +1,22 @@
+# Cluster API v1.8 compared to v1.9
+
+This document provides an overview over relevant changes between Cluster API v1.8 and v1.9 for
+maintainers of providers and consumers of our Go API.
+
+## Go version
+
+- The Go version used by Cluster API is Go 1.22.x
+
+## Changes by Kind
+
+### Deprecation
+
+### Removals
+
+### API Changes
+
+### Other
+
+### Suggested changes for providers
+
+- The Errors package was created when capi provider implementation was running as machineActuators that needed to vendor core capi to function. There is no usage recommendations today and its value is questionable since we moved to CRDs that inter-operate mostly via conditions. Instead we plan to drop the dedicated semantic for terminal failure and keep improving Machine lifecycle signal through conditions. Therefore the Errors package [has been deprecated in v1.8](https://github.com/kubernetes-sigs/cluster-api/issues/10784). It's recommented to remove any usage of the currently exported variables.
```

Comment:
> The Errors package was created when capi provider implementation was running as machineActuators that needed to vendor core capi to function. There is no usage recommendations today

Just to avoid misunderstandings (and to ensure we're talking about the same thing)

MachineStatusError & ClusterStatusError are types used in our API and they are both used to read the corresponding fields from InfrastructureCluster & InfrastructureMachine (e.g.: https://github.com/kubernetes-sigs/cluster-api/blob/4714c48b6221b67cf4a8955be3e5f79352dc7532/internal/controllers/machine/machine_controller_phases.go#L105-L111)

So as of right now, they are even part of our contract (vs. "no usage recommendation").
But this will be deprecated with https://github.com/kubernetes-sigs/cluster-api/pull/10897 and then removed with v1beta2.

Correct?

Reply:
(cc @fabriziopandini)

Reply:
Sounds good!

Reply:
+1